lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -169,4 +194,37 @@ public void submitTask(final String jobId, final String taskId, final URI jobJar
   public String getAzureBatchJobId() {
     return System.getenv(AZ_BATCH_JOB_ID_ENV);
   }
+
+  private TaskContainerSettings getTaskContainerSettings() {
+    if (this.containerRegistry == null) {
+      return null;
+    }
+
+    StringBuilder portMappings = new StringBuilder();
+    Iterator<Integer> iterator = this.portProvider.iterator();
+    while (iterator.hasNext()) {
+      Integer port = iterator.next();
+      portMappings.append(String.format("-p %d:%d ", port, port));
+    }
+
+    return new TaskContainerSettings()
+        .withRegistry(this.containerRegistry)
+        .withImageName(this.containerRegistryProvider.getContainerImageName())
+        .withContainerRunOptions(
+            String.format(
+                "-dit --env HOST_IP_ADDR_PATH=%s %s",
+                this.commandBuilder.getIpAddressFilePath(),
+                portMappings));
+  }
+
+  private JobPreparationTask getJobPreparationTask() {
+    if (this.containerRegistry == null) {
+      return null;
+    }
```

Comment:
Instead of checking some null for being set, it's more explanatory if you create a method like "areContainersEnabled"?